presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardCleaner.java

```diff
@@ -275,9 +267,35 @@ private synchronized void runBackupCleanup()
         }
     }
 
+    private synchronized void waitJitterTime()
+    {
+        try {
+            // jitter to avoid overloading database
+            long interval = this.localCleanerInterval.roundTo(SECONDS);
+            SECONDS.sleep(ThreadLocalRandom.current().nextLong(1, interval));
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private synchronized void runLocalCleanupAtStartup()
+    {
+        try {
+            Set<UUID> local = getLocalShards();
```

Comment:
I'd get the list outside of scheduled method in a blocking way. That way it's guaranteed to have only the files you care about. That's what David originally suggested.